src/plugins/data/public/search/fetch/handle_warnings.tsx

```diff
@@ -21,33 +23,56 @@ import {
 } from '../types';
 import { extractWarnings } from './extract_warnings';
 
+const getDebouncedWarning = () => {
+  const addWarning = () => debounce(getNotifications().toasts.addWarning, 10000, { leading: true });
```

Comment:
Does this mean that regardless of the deduping, we'll still only show a max of one warning every 10 seconds?

Reply:
The debounce is applied for a given `debounceKey`. Different debounceKeys within 10s would result in multiple warnings. The same debounceKey within 10s will result in only one warning, past 10s an additional warning is shown.